catalog/core/catalog-core-transform-impl/src/test/java/org/codice/ddf/catalog/transform/impl/TransformImplTest.java

```diff
@@ -0,0 +1,497 @@
+import org.codice.ddf.catalog.transform.MultiMetacardTransformer;
+import org.codice.ddf.catalog.transform.Transform;
+import org.codice.ddf.catalog.transform.TransformResponse;
+import org.codice.ddf.platform.util.uuidgenerator.UuidGenerator;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+import org.mockito.ArgumentCaptor;
+
+@SuppressWarnings("unchecked")
+public class TransformImplTest {
+
+  @Rule public TemporaryFolder temporaryFolder = new TemporaryFolder();
+
+  private File tempFile;
+
+  private Metacard metacard;
+
+  private TransformerLocator transformerLocator;
+
+  private Transform transform;
+
+  private Supplier<String> idSupplier;
+
+  @Before
+  public void setup() throws IOException {
+    tempFile = temporaryFolder.newFile("tempFile");
+    metacard = mock(Metacard.class);
+    transformerLocator = mock(TransformerLocator.class);
+    UuidGenerator uuidGenerator = mock(UuidGenerator.class);
+    transform = new TransformImpl(transformerLocator, uuidGenerator);
+    idSupplier = mock(Supplier.class);
+    when(idSupplier.get()).thenReturn(null);
+  }
+
+  @Test
+  public void testTransformSourceResponseByTransformerId() throws CatalogTransformerException {
+
+    BinaryContent mockBinaryContent = mock(BinaryContent.class);
+
+    QueryResponseTransformer transformer = mock(QueryResponseTransformer.class);
+    when(transformer.transform(any(SourceResponse.class), any(Map.class)))
+        .thenReturn(mockBinaryContent);
+
+    when(transformerLocator.findQueryResponseTransformers(eq("xyz")))
+        .thenReturn(Collections.singletonList(transformer));
+
+    BinaryContent binaryContent =
+        transform.transform(mock(SourceResponse.class), "xyz", Collections.emptyMap());
+
+    assertThat(binaryContent, is(mockBinaryContent));
+  }
+
+  @Test
+  public void testTransformSourceResponseByMimeType()
+      throws CatalogTransformerException, MimeTypeParseException {
+
+    BinaryContent mockBinaryContent = mock(BinaryContent.class);
+
+    QueryResponseTransformer transformer = mock(QueryResponseTransformer.class);
+    when(transformer.transform(any(SourceResponse.class), any(Map.class)))
+        .thenReturn(mockBinaryContent);
+
+    when(transformerLocator.findQueryResponseTransformers(any(MimeType.class)))
+        .thenReturn(Collections.singletonList(transformer));
+
+    BinaryContent binaryContent =
+        transform.transform(
+            mock(SourceResponse.class), new MimeType("text/xml"), Collections.emptyMap());
+
+    assertThat(binaryContent, is(mockBinaryContent));
+  }
+
+  @Test
+  public void testIsMetacardTransformerIdValidBadId() throws CatalogTransformerException {
+
+    BinaryContent binaryContent = mock(BinaryContent.class);
+
+    MultiMetacardTransformer transformer = mock(MultiMetacardTransformer.class);
+    when(transformer.transform(any(List.class), any(Map.class)))
```

Comment:
Is this stubbing necessary for the test?

Reply:
removed